src/Module.cpp

```diff
@@ -535,6 +535,10 @@ void Module::compile(const std::map<OutputFileType, std::string> &output_files)
         debug(1) << "Module.compile(): stmt_html " << output_files.at(OutputFileType::stmt_html) << "\n";
         Internal::print_to_html(output_files.at(OutputFileType::stmt_html), *this);
     }
+    if (contains(output_files, OutputFileType::stmt_viz)) {
+        debug(1) << "Module.compile(): stmt_viz " << output_files.at(OutputFileType::stmt_viz) << "\n";
+        Internal::print_to_viz(output_files.at(OutputFileType::stmt_viz), *this);
```

Comment:
Unclear why these lines were added, this code is already present on line 621 https://github.com/halide/Halide/blob/929b5d7ce952ec12753c0f7637d67375a6c6c5eb/src/Module.cpp#L621

Since the features displays the compiled assembly, we run it after the assembly flag has been processed.